source/fe/mapping_cartesian.cc

```diff
@@ -37,6 +37,32 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+DeclExceptionMsg(
+  ExcCellNotCartesian,
+  "You are using MappingCartesian, but the incoming cell is not Cartesian.");
+
+
+
+/**
+ * Return whether the incoming cell is a Cartesian. This is determined by
+ * checking if the smallest BoundingBox that encloses the cell has the same
+ * vertices as the cell itself.
+ */
+template <class CellType>
+bool
+is_cartesian(const CellType &cell)
+{
+  const auto bounding_box = cell->bounding_box();
```

Comment:
No sure if going via bb is right way. Wouldn't it be easier to go through each direction and compare the points?

Reply:
I don't know. I can't figure out something easier right now, at least. I think I would just end up with something very similar to what is already in the `bounding_box()` function:

https://www.dealii.org/developer/doxygen/deal.II/tria__accessor_8cc_source.html#l01616

Reply:
Let's keep it like this!